### PR TITLE
Allow External Urls / Parse Remote Markdown files

### PR DESCRIPTION
**Pull request by kzap**
With this update you can link to absolute URLs in your Table of Contents and it will link to it without trying to parse it.

More importantly, if you have a markdown file somewhere else on the internet or in another GIthub repository you can use it in your Table of Contents, you can even include `username:password` or `token` in the URL for private repos. Also works with Bitbucket URLs:

screenshot 2017-08-13 14 57 46

Sample Table of Contents `docs/README.md`
```
* [docpress](../README.md)
* [External Link to Github](https://github.com)
* [Include Remote Github Markdown]
(https://github.com/docpress/docpress/blob/master/README.md)
* [Include Another Remote Github Markdown](https://github.com/docpress/docpress/blob/master/docs/showcase.md)
* [Include Remote Bitbucket Markdown](https://bitbucket.org/MartinFelis/love-android-sdl2/src/a1aca69aa453ad41f45234c8d7739136637acc72/README.md)
```

### CODE REVIEW

File: lib/fix_html.js [hunk 1]
```diff
@@ -56,17 +57,37 @@ function fixReferences ($, fname, sources, files, page) {
...
-    if (!sources[target]) {
-      throw new Error(`${base}: Unknown reference '${origUrl}'`)
-    }
+      // remove username/password
+      parsedUrl.set('username', '')
+      parsedUrl.set('password', '')
+
+      let baseSource = parsedUrl.href
+      baseSource = baseSource.split('/')
+      baseSource.splice(-1)
```
Comment (knownasilya) [inline lib/fix_html.js]: can we use `slice` instead, and assign the value.

Comment (kzap) [inline lib/fix_html.js]: I changed this to use dirname() instead

**Comment by kzap** [comment]
Improved the error checking, fixed linter errors and also made remote markdown files link to each other if they are both in the Table of Contents. Example:
```
* [Include Remote Github Markdown]
(https://github.com/docpress/docpress/blob/master/README.md)
* [Include Another Remote Github Markdown](https://github.com/docpress/docpress/blob/master/docs/showcase.md)```

File: lib/fix_html.js [hunk 2]
```diff
@@ -56,17 +58,60 @@ function fixReferences ($, fname, sources, files, page) {
...
     }
 
-    // Get the target source file it points to (eg, `docs/usage.md`).
-    var target = getTarget(origUrl, base)
-
-    // Ensure that it's available.
-    if (!sources[target]) {
-      throw new Error(`${base}: Unknown reference '${origUrl}'`)
+    // if absolute markdown file, make link relative to MD file
+    if (page.absolute) {
```
Comment (knownasilya) [inline lib/fix_html.js]: Currently this isn't tested. If you are still interested, do you think you could add a test? I rebased your branch and updated the tests to pass, but this brings down the coverage too much.

**Comment by kzap** [comment]
Thank you for updating this but currently I dont have the time to commit to
updating this.

**Comment by knownasilya** [comment]
@rstacruz I'll add the tests 👍

**Pull request closed without merging**